### PR TITLE
Spec attributes refactoring (fix #219)

### LINKED ISSUE #219: Spec attrs refactoring
to have it visible in docs and make it properly working, we need to:
- [ ] make all "hidden dict" attributes normal class attributes (they should be named and accessible the same (except we couldn't use spec["bla"] anymore)
- [ ] make Provides, Requires and BuildRequires, build_required_files and required_files a sets
- [ ] make sure all tests pass and fix the ones in test_spec.py

### PR DESCRIPTION
**Pull request by LukasSlouka**
Read comments in #219

### CODE REVIEW

File: rpg/plugins/misc/files_to_pkgs.py [hunk 1]
```diff
@@ -16,12 +16,14 @@ def _resolve(files, query):
                                 .format(_file))
 
         def _uniq(_list):
```
Comment (jsilhan) [inline rpg/plugins/misc/files_to_pkgs.py]: remove the `_uniq` method then, please.

**Comment by jsilhan** [comment]
You understood it as I though. Do we need `__getattr__` and `__setattr__`?

**Comment by LukasSlouka** [comment]
Those methods were there, I just rewrote them, but I think we need them. Changelog fixed and `_uniq` removed.

**Comment by LukasSlouka** [comment]
Hmm `__getattr__` is redundant after all, but `__setattr__` is converting some types so it should stay

**Comment by jsilhan** [comment]
If the tests pass without `__setattr__`, then remove it, please. IMO it's there from archaic age.

**Comment by LukasSlouka** [comment]
That is the thing, they do not pass without it, but I can try to find workaround.

```python
self.spec.build = ["cmake .", "make"]
self.assertTrue(isinstance(self.spec.build, Command))
self.spec.install = "make install"
self.assertTrue(isinstance(self.spec.install, Command))

self.assertRaises(AttributeError, setattr, self.spec, "bla", "a")
self.assertRaises(TypeError, setattr, self.spec, "Name", 3)
self.assertRaises(TypeError, setattr, self.spec, "description", Command("c"))
self.assertRaises(TypeError, setattr, self.spec, "Name", ["a", "b"])
```

These asserts casue the failure without it

**Comment by jsilhan** [comment]
replace these and all tests with `setattr` erase, please
```python
self.spec.build = Command(["cmake .", "make"])
...
self.spec.install = Command("make install")
```

**Comment by LukasSlouka** [comment]
Basically it is doing type control and implicit conversion if needed. If we don't want that kind of behavior, `__setattr__` can be removed, but also following tests loose their purpose and should be removed: 

```python
self.assertTrue(isinstance(self.spec.build, Command))
self.assertTrue(isinstance(self.spec.build, Command))
self.assertTrue(isinstance(self.spec.install, Command))
self.assertTrue(isinstance(self.spec.check, Command))
self.assertTrue(isinstance(self.spec.Requires, set))
self.assertTrue(isinstance(self.spec.Requires, set))
self.assertRaises(AttributeError, getattr, self.spec, "bla")
self.assertRaises(TypeError, setattr, self.spec, "Name", 3)
self.assertRaises(TypeError, setattr, self.spec, "description", Command("c"))
self.assertRaises(TypeError, setattr, self.spec, "Name", ["a", "b"])
```

**Comment by LukasSlouka** [comment]
I can replace 

```python
self.assertRaises(AttributeError, setattr, self.spec, "bla", "a")
```

with

```python
self.assertRaises(AttributeError, getattr, self.spec, "bla")
```

**Comment by jsilhan** [comment]
Merged, thanks

**Comment by cibo94** [comment]
```
Found violations: rpg/plugins/lang/c.py
	E128 @ 29:29 - continuation line under-indented for visual indent
Found violations: rpg/plugins/misc/files_to_pkgs.py
	E501 @ 21:80 - line too long (91 > 79 characters)
	E501 @ 23:80 - line too long (97 > 79 characters)
	E128 @ 24:50 - continuation line under-indented for visual indent
Found violations: tests/unit/test_spec.py
	W293 @ 44:1 - blank line contains whitespace
	E303 @ 46:5 - too many blank lines (2)
```

File: rpg/spec.py [hunk 2]
```diff
@@ -44,126 +44,79 @@ class Subpackage(dict):
...
+    build = Command()
+    pre = Command()
+    install = Command()
+    check = Command()
+    post = Command()
+    preun = Command()
+    postun = Command()
+    pretrans = Command()
+    posttrans = Command()
+    clean = Command()
```
Comment (cibo94) [inline rpg/spec.py]: I think this all tags shouldn't be class but instance variables:

```python
def __init__(self):
    self.Name = ""
    ...
```

**Comment by LukasSlouka** [comment]
I am back from France, will take a look at the flake8 errors if you want. Don't know about the tags, does the project use multiple instances of spec class? And I am not sure how does the sphinx react to instance variables in documentation. I will test it later today.

**Comment by cibo94** [comment]
no I have done it in one massive commit. And spec class may be instancialized multiple times for instance in tests ... maybe because of that this #223 is failing

**Comment by LukasSlouka** [comment]
Btw if tags were instance attributes, sphinx would not be able to document them unless they would have attached docstring to them like this

```python
class SomeClass(object):

    _init__(self, par1, par2):
        self.attr1 = val1 #: initial value: val1
        self.attr1 = val2 #: initial value: val2
``` 

or like this

```python
class SomeClass(object):
    """    
    Description for class

    :ivar attr1: initial value: val1
    :ivar attr1: initial value: val2
    """
    _init__(self, par1, par2):
        self.attr1 = val1
        self.attr1 = val2
```

**Comment by jsilhan** [comment]
go with the first version, please. Otherwise we'll forget to update them in class docstring.

**Pull request closed without merging**